pkg/virt-controller/watch/vmi.go

```diff
@@ -2216,30 +2216,31 @@ func (c *VMIController) handleDynamicInterfaceRequests(vmi *virtv1.VirtualMachin
 }
 
 func (c *VMIController) updateInterfaceStatus(vmi *virtv1.VirtualMachineInstance, pod *k8sv1.Pod) error {
-	originalVMIStatus := vmi.Status.DeepCopy().Interfaces
-	indexedVMIStatus := indexIfaceStatus(originalVMIStatus)
+	indexedVMIStatus := indexIfaceStatus(vmi.Status.Interfaces)
 
-	var newStatus []virtv1.VirtualMachineInstanceNetworkInterface
 	ifaceNamingScheme := namescheme.CreateNetworkNameScheme(vmi.Spec.Networks)
 	indexedPodIfaces := nonDefaultMultusNetworksIndexedByIfaceName(pod)
 	for _, network := range vmi.Spec.Networks {
-		status := virtv1.VirtualMachineInstanceNetworkInterface{Name: network.Name}
-		if existingIfaceStatus, ok := indexedVMIStatus[network.Name]; ok {
-			status = existingIfaceStatus
+		if _, alreadyPresentInVMIStatus := indexedVMIStatus[network.Name]; alreadyPresentInVMIStatus {
+			continue
 		}
 
 		podIfaceName, wasFound := ifaceNamingScheme[network.Name]
 		if !wasFound {
 			return fmt.Errorf("could not find the pod interface name for network [%s]", network.Name)
 		}
 
-		if _, found := indexedPodIfaces[podIfaceName]; found {
-			status.PodConfigDone = true
+		if _, isPodInterfaceAvailable := indexedPodIfaces[podIfaceName]; !isPodInterfaceAvailable {
+			continue
```

Comment:
This has an assumption that the `PodConfigDone` is always set and never `false` for existing interfaces.
While this is correct now, it depends on many factors that may change without realizing this will cause a problem here.

Said that, we could handle this as a follow up.